service/migration_manager.cc

```diff
@@ -1253,6 +1253,10 @@ future<schema_ptr> migration_manager::get_schema_for_write(table_schema_version
 }
 
 future<> migration_manager::sync_schema(const replica::database& db, const std::vector<gms::inet_address>& nodes) {
+    if (!_enable_schema_pulls) {
+        co_await _group0_barrier.trigger(_as);
+        co_return;
+    }
```

Comment:
Isn't this a bit of a pessimization that may increase latency?
This happens in `repair_range`, so for every token range IIUC. Could be important.

Below, we start with comparing schema versions, and only do pulls if the versions are different.
Perhaps we should do the barrier only if there is a difference too?

Reply:
I wonder how much overhead group0 barrier actually has, in the steady state. Checking schema versions involves sending RPCs to each involved node, so it's not free either. The raft read barrier needs to contact a quorum. On the surface looks similar. Is it heavier?

I measured the speed of repair on bootstrap in dev mode, and for system_distributed keyspace I get 16s before the patch and 22s after the patch, which would indicate that it is indeed heavier. This is the worst case, because ranges are empty and there's 3000 of them.

Reply:
There's round-trip latency from barrier caller to the reader, then round-trip latency from leader to quorum, then there's waiting for the index we got from the leader to be applied -- this wait should be instant if there are no going group 0 operations, but there could potentially be some ongoing group 0 operations not necessarily related to schema.